packages/contracts/src/extensions/UniswapExtension.sol

```diff
@@ -8,28 +8,36 @@ import {EmailWalletCore} from "../EmailWalletCore.sol";
 import {TokenRegistry} from "../utils/TokenRegistry.sol";
 import "../interfaces/Types.sol";
 
+import "./PoolFinder.sol";
+
+
 contract UniswapExtension is Extension {
     EmailWalletCore public core;
     ISwapRouter public router;
     TokenRegistry public tokenRegistry;
+    PoolFinder public poolFinder;
 
     // For this example, we will set the pool fee to 0.3%.
     uint24 public constant poolFee = 3000;
+    // For this example, we will set the maximum slippage to 0.5%.
+    uint24 public constant slippageBasisPoints = 50;
 
     mapping(string => address) public addressOfNFTName;
 
-    string[][] public templates = new string[][](1);
+    string[][] public templates = new string[][](2);
 
     modifier onlyCore() {
         require((msg.sender == address(core)) || (msg.sender == address(core.unclaimsHandler())), "invalid sender");
         _;
     }
 
-    constructor(address coreAddr, address _tokenReg, address _router) {
+    constructor(address coreAddr, address _tokenReg, address _router, address _factory) {
         core = EmailWalletCore(payable(coreAddr));
         tokenRegistry = TokenRegistry(_tokenReg);
         router = ISwapRouter(_router);
         templates[0] = ["Swap", "{tokenAmount}", "to", "{string}"];
+        templates[1] = ["Swap", "{tokenAmount}", "to", "{string}", "under", "{uint}", "sqrt price limit"];
```

Comment:
I think the command should be more user friendly 
`Swap 20 ETH to USDC with 5% slippage" (slippage instead of sqrt price) can we do that?

Reply:
> I think the command should be more user friendly
`Swap 20 ETH to USDC with 5% slippage" (slippage instead of sqrt price) can we do that?

I agree with that point. You can use {amount} template to represent decimals, e.g., 5.12 in the subject. Its corresponding parameter is provided as the unsigned integer with decimal size 18, i.e., 5.12 * 10^18.

Reply:
Yes using {amount} - sorry for the bad example with percentage.